File: src/main/java/br/com/dusty/dkits/util/InventoryUtils.java
package br.com.dusty.dkits.util;

import br.com.dusty.dkits.util.text.Text;
import br.com.dusty.dkits.util.text.TextColor;
import org.bukkit.Bukkit;
import org.bukkit.Material;
import org.bukkit.entity.Player;
import org.bukkit.inventory.Inventory;
import org.bukkit.inventory.ItemStack;
import org.bukkit.inventory.PlayerInventory;

public class InventoryUtils {
	
	public static final ItemStack SOUP = new ItemStack(Material.MUSHROOM_SOUP);
	
	public static final ItemStack RED_MUSHROOMS = new ItemStack(Material.RED_MUSHROOM, 64);
	public static final ItemStack BROWN_MUSHROOMS = new ItemStack(Material.BROWN_MUSHROOM, 64);
	public static final ItemStack BOWLS = new ItemStack(Material.BOWL, 64);
	
	private static final String SOUPS_TITLE = Text.of("Sopas").color(TextColor.GOLD).toString();
	private static final String RECRAFT_TITLE = Text.of("Recraft").color(TextColor.GOLD).toString();
	
	private static final ItemStack[] ARMOR_FULLIRON = new ItemStack[]{new ItemStack(Material.IRON_HELMET),
	                                                                  new ItemStack(Material.IRON_CHESTPLATE),
	                                                                  new ItemStack(Material.IRON_LEGGINGS),
	                                                                  new ItemStack(Material.IRON_BOOTS)};
	
	public static Inventory soups(Player player) {
		Inventory inventory = Bukkit.createInventory(player, 54, SOUPS_TITLE);
		
		for(int i = 0; i < 54; i++){
			inventory.setItem(i, SOUP);
		}
		
		return inventory;
	}
	
	public static Inventory recraft(Player player) {
		Inventory inventory = Bukkit.createInventory(player, 54, RECRAFT_TITLE);
		
		for(int i = 0; i < 18; i++){
			inventory.setItem(i * 3, RED_MUSHROOMS);
		}
		
		for(int i = 0; i < 18; i++){
			inventory.setItem((i * 3) + 1, BROWN_MUSHROOMS);
		}
		
		for(int i = 0; i < 18; i++){
			inventory.setItem((i * 3) + 2, BOWLS);
		}
		
		return inventory;
	}
	
	public static void addItemStacks(Inventory inventory, ItemStack[] itemStacks) {
		int i = 0;
		
		while(inventory.getItem(i) != null)
			i++;
		
		for(ItemStack itemStack : itemStacks)
			inventory.setItem(i++, itemStack);
	}
	
	/**
	 * Aplica um {@link ItemStack}[] contendo, nessa ordem, 'helmet', 'chestplate', 'leggings' e 'boots' como armadura de um {@link Player}.
	 *
	 * @param player
	 * @param itemStacks
	 */
	public static void setArmor(Player player, ItemStack[] itemStacks) {
		PlayerInventory playerInventory = player.getInventory();
		playerInventory.setHelmet(itemStacks[0]);
		playerInventory.setChestplate(itemStacks[1]);
		playerInventory.setLeggings(itemStacks[2]);
		playerInventory.setBoots(itemStacks[3]);
	}
}

File: src/main/kotlin/br/com/dusty/dkits/util/inventory/InventoryUtils.kt
package br.com.dusty.dkits.util.inventory

import br.com.dusty.dkits.util.rename
import br.com.dusty.dkits.util.text.Text
import br.com.dusty.dkits.util.text.TextColor
import org.bukkit.Bukkit
import org.bukkit.Material
import org.bukkit.entity.Player
import org.bukkit.inventory.Inventory
import org.bukkit.inventory.ItemStack

fun Inventory.addItemStacks(itemStacks: Array<ItemStack?>) {
	var i = 0

	while (this.getItem(i) != null) i++

	for (itemStack in itemStacks) this.setItem(i++, itemStack)
}

/**
 * Aplica um [ItemStack][] contendo, nessa ordem, 'helmet', 'chestplate', 'leggings' e 'boots' como armadura de um [Player].
 *
 * @param player
 * @param itemStacks
 */
fun Player.setArmor(itemStacks: Array<ItemStack?>) {
	val playerInventory = this.inventory

	playerInventory.helmet = itemStacks[0]
	playerInventory.chestplate = itemStacks[1]
	playerInventory.leggings = itemStacks[2]
	playerInventory.boots = itemStacks[3]
}

fun Inventory.fill(backButton: Boolean): Inventory {
	for (i in 0 until this.size) this.setItem(i, InventoryUtils.BACKGROUND)

	if (backButton) this.setItem(0, InventoryUtils.BUTTON_BACK)

	return this
}

object InventoryUtils {

	val SOUP = ItemStack(Material.MUSHROOM_SOUP)

	val RED_MUSHROOMS = ItemStack(Material.RED_MUSHROOM, 64)
	val BROWN_MUSHROOMS = ItemStack(Material.BROWN_MUSHROOM, 64)
	val BOWLS = ItemStack(Material.BOWL, 64)

	val SOUPS_TITLE = Text.of("Sopas").color(TextColor.GOLD).toString()
	val RECRAFT_TITLE = Text.of("Recraft").color(TextColor.GOLD).toString()

	val ARMOR_FULL_IRON = arrayOf(ItemStack(Material.IRON_HELMET), ItemStack(Material.IRON_CHESTPLATE), ItemStack(Material.IRON_LEGGINGS), ItemStack(Material.IRON_BOOTS))

	val BACKGROUND = ItemStack(Material.STAINED_GLASS_PANE, 1, 1.toShort(), 7.toByte()).rename(" ")

	val BUTTON_BACK = ItemStack(Material.CARPET, 1, 1.toShort(), 14.toByte()).rename(Text.negativeOf("Voltar").toString())

	fun soups(player: Player): Inventory {
		val inventory = Bukkit.createInventory(player, 54, SOUPS_TITLE)

		for (i in 0 .. 53) inventory.setItem(i, SOUP)

		return inventory
	}

	fun recraft(player: Player): Inventory {
		val inventory = Bukkit.createInventory(player, 54, RECRAFT_TITLE)

		for (i in 0 .. 17) {
			inventory.setItem(i * 3, RED_MUSHROOMS)
			inventory.setItem(i * 3 + 1, BROWN_MUSHROOMS)
			inventory.setItem(i * 3 + 2, BOWLS)
		}

		return inventory
	}
}
